### PR TITLE
NH-59205 PoC layer-python includes solarwinds-apm

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
WIP

### CODE REVIEW

File: python/src/otel/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-ARG runtime=python3.11
+ARG runtime=python3.8
```
Comment (jerrytfleung) [inline python/src/otel/Dockerfile:1]: Why do we set the runtime back to python3.8?

File: python/src/otel/otel_sdk/requirements.txt [hunk 2]
```diff
@@ -1,3 +1,5 @@
-opentelemetry-exporter-otlp-proto-http==1.19.0
-opentelemetry-distro==0.40b0
-opentelemetry-instrumentation-aws-lambda==0.40b0
+opentelemetry-exporter-otlp==1.20.0
+opentelemetry-exporter-otlp-proto-http==1.20.0
+opentelemetry-exporter-otlp-proto-grpc==1.20.0
+opentelemetry-distro==0.41b0
```
Comment (jerrytfleung) [inline python/src/otel/otel_sdk/requirements.txt]: Do we need opentelemetry-distro? I was thinking solarwinds-apm itself is a distro

File: python/src/otel/Dockerfile (hunk 1 above)
Comment (tammy-baylis-swi) [reply python/src/otel/Dockerfile:1]: My first lambda test functions were Python 3.8 so I made this working change to accommodate and use the correct extension .so inside the layer-python used in the lambda environment.

Moving forward we'll need to support Python 3.8 to 3.11. Otel does proper builds using [template](https://github.com/solarwinds/opentelemetry-lambda/blob/NH-59205-apm-python-custom-layer/python/src/template.yml) and Makefile on GH I think? Not with this local setup.

File: python/src/otel/otel_sdk/requirements.txt (hunk 2 above)
Comment (tammy-baylis-swi) [reply python/src/otel/otel_sdk/requirements.txt]: Nope not needed, removed in [615f34e](https://github.com/solarwinds/opentelemetry-lambda/pull/1/commits/615f34e0e128f68e2add6be38abb15a48df4b4a5). Created a test layer-python to be sure, and works without it. I think this dep is leftover from when I tried to build layer-python as vanilla distro, without APM Python.

File: python/src/otel/otel_sdk/requirements.txt [hunk 3]
```diff
@@ -1,3 +1,3 @@
-opentelemetry-exporter-otlp-proto-http==1.19.0
-opentelemetry-distro==0.40b0
-opentelemetry-instrumentation-aws-lambda==0.40b0
+opentelemetry-exporter-otlp==1.20.0
```
Comment (tammy-baylis-swi) [inline python/src/otel/otel_sdk/requirements.txt:1]: Note to self, this seems to be needed by APM Python layer to use `otlp` default entry point for `EXPORTER` config.

Why it has to be added here, I don't know yet. The vanilla `opentelemetry-distro` [packages opentelemetry-exporter-otlp](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/3478831838ce54a972f7434ae6098f9bb28dc494/opentelemetry-distro/pyproject.toml#L33) which may be why `main` branch doesn't need this dep. But if I add it as a dep to APM Python's setup.cfg and don't add it to this txt for layer building, lambda environment gets angry with `pkg_resources.DistributionNotFound: The 'opentelemetry-exporter-otlp==1.20.0' distribution was not found and is required by the application`. It cites these lines

* https://github.com/open-telemetry/opentelemetry-python-contrib/blob/3478831838ce54a972f7434ae6098f9bb28dc494/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/_load.py#L40
* https://github.com/open-telemetry/opentelemetry-python-contrib/blob/3478831838ce54a972f7434ae6098f9bb28dc494/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py#L36

**Comment by tammy-baylis-swi** [comment]
Closing this because APM Python layer builds are now done by APM Python repo.

**Pull request closed without merging**